filebeat/docs/modules/googlecloud.asciidoc

```diff
@@ -0,0 +1,75 @@
+////
+This file is generated! See scripts/docs_collector.py
+////
+
+[[filebeat-module-googlecloud]]
+[role="xpack"]
+
+:modulename: googlecloud
+:has-dashboards: false
+
+== Google Cloud module
+
+beta[]
+
+This is a module for Google Cloud logs. It supports reading VPC flow logs that
+have been exported from Stackdriver to a Google Pub/Sub topic sink.
+
+include::../include/what-happens.asciidoc[]
+
+include::../include/running-modules.asciidoc[]
+
+include::../include/configuring-intro.asciidoc[]
+
+:fileset_ex: vpcflow
+
+include::../include/config-option-intro.asciidoc[]
+
+[float]
+==== `vpcflow` fileset settings
+
+Example config:
+
+[source,yaml]
+----
+- module: googleclcoud
+  vpcflow:
+    enabled: true
+    var.project_id: my-gcp-project-id
+    var.topic: googlecloud-vpc-flowlogs
+    var.subscription_name: filebeat-googlecloud-vpc-flowlogs-sub
+    var.credentials_file: ${path.config}/gcp-service-account-xyz.json
+----
+
+include::../include/var-paths.asciidoc[]
+
+*`var.project_id`*::
+
+Google Cloud project ID.
+
+*`var.topic`*::
+
+Google Cloud Pub/Sub topic name.
+
+*`var.subscription_name`*::
+
+Google Cloud Pub/Sub topic subscription name. If the subscription does not
+exist it will be created.
+
+*`var.credentials_file`*::
+
+Path to a JSON file containing the credentials and key used to subscribe.
```

Comment:
Not for this PR, but eventually we'll want to support the secrets store for this?

Reply:
Yeah, the underlying input supports the keystore via `credentials_json`, where you can embed the whole json blob as string/bytes in the config this way the keystore is supported. But I didn't expose param it in the module, but probably should in the future.